### DISCUSSION TITLE
MastoAPI/Soapbox integration possibilities

### DISCUSSION THREAD

**Discussion by plantroon** [Automattic/wordpress-activitypub#282] [Ideas]
Would it be possible to implement such extent of ActivityPub and MastoAPI that majority of features would be available so this would be on par with Mastodon and other popular ActivityPub implementations? The main desire is to have a website at example.com and a fediverse handle such as user@example.com to which I could connect via the mobile apps meant for ActivityPub (Tusky, AndStatus..) or via Soapbox (at soapbox.example.com). Is there any demand for this? It seems like a good idea, though maybe only to me.

Friendica is written in PHP and compatible with a lot of protocols, if that would be of any help. I was actually trying to specify nginx locations corresponding to ActivityPub (Friendica) endpoints and Wordpress endpoints and pass them to respective backends, but it screams with trouble - and in such case I lose the ability to use this plugin.

**Comment by pfefferle** [comment]
Hey @plantroon thanks for your suggestion! I would love to add a Client/Server API to support Apps, but I would prefer ActivityPub over the proprietary API of mastodon (what is used by tusky).

In the end it is a matter of time! It is still a spare time project and there is a lot on my list before working on such kind of an API.

**Reply by huckabarry** [reply]
Just wanted to point out that there is a plugin now called Mastodon Apps. It's still in its infancy, but it's working as of yesterday on a bare bones level. Needs work but going in the right direction. Some clients work better than others.
